### DISCUSSION TITLE
Modify URL before copying

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [gdh1995/vimium-c#1076] [Q&A] [answered]
How do I go about using a pattern I've defined in the config?

**Comment by gdh1995** [comment]
Syntax can be seen in https://github.com/gdh1995/vimium-c/wiki/Substitute-URLs-and-text-during-commands .

And in v1.99+, `sed=<prefix-keys>` should work exactly the same as `sedKeys=...`, so you may write:
```
# the beginning character of "_" means a single long sed key (at most 8 characters)
map <c-s-c> autoCopy sed="_rule1"
```
and 
```
_rule1/<regexp-pattern>/<substitution-results>/<flags>...
```

**Reply by limaceous-bushwhacker** [reply]
Thank you, didn't know about `autoCopy`. I'm trying to copy video ID from a YouTube "watch" page. What'd be the best way to go about it? I reckon `/watch?v=([w-]{11})` regex should work. Can't seem to get it to work ATM.

```
unmap yy
map yy autoCopy sed="3"
```

```
3@youtube.com/watch?v=([\w-]{11})@$1@,matched
```

**Reply by gdh1995** [reply, marked as answer]
You regexp has some errors.

Please try:
```
map yy copyCurrentUrl sed="_vid"
```
```
_vid@\byoutube\.com/watch\?v=([\w-]{10,})@$1@,matched
```

**Reply by limaceous-bushwhacker** [reply]
Ah, thanks! Unrelated but can we `map` within `env` directive or does it only support `run`/`runKey`?

**Reply by gdh1995** [reply]
Of course you may combine `env` and `map` - `run key xxx ***` is just a shortcut of `map key runKey keys="xxx" ***`
